app/uk/gov/hmrc/merchandiseinbaggagefrontend/forms/SearchGoodsFormProvider.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggagefrontend.forms
+
+import javax.inject.Inject
+import play.api.data.Form
+import play.api.data.Forms.mapping
+import uk.gov.hmrc.merchandiseinbaggagefrontend.forms.mappings.Mappings
+import uk.gov.hmrc.merchandiseinbaggagefrontend.model.declaration.CategoryQuantityOfGoods
+
+class SearchGoodsFormProvider @Inject() extends Mappings {
```

Comment:
In this case is there benefit in having a Provider rather than just a Form?

Reply:
I'm just not clear of the benefit of the provider and it is more succinct just to have the form as an object.

I think at some point we'll refactor to be consistent one way or the other. Doesn't stop this being merged now though...

Reply:
I think by having it this way I can extend our Mappings trait, whereas that isn't possible with an Object.

In your travellerDetails form, you use Play's nonEmptyText, but that doesn't give you the opportunity to specify an errorKey, whereas our custom implementation does. There also aren't any tests for you travellerDetails form either 🧐

Reply:
True I didn't add a form spec but I thought all the functionality was tested in the controller spec...

Do we need to specify an error key? Is there some error handling behaviour that's missing in the TravellerDetailsPage?